source/fe/fe_values.cc

```diff
@@ -16,8 +16,10 @@
 #include <deal.II/base/array_view.h>
 #include <deal.II/base/memory_consumption.h>
 #include <deal.II/base/multithread_info.h>
+#include <deal.II/base/numbers.h>
 #include <deal.II/base/quadrature.h>
 #include <deal.II/base/signaling_nan.h>
+#include <deal.II/differentiation/ad.h>
```

Comment:
Note that if we use the unifying header then we avoid having to use this (potentially increasing) list of headers:
```
#include <deal.II/differentiation/ad/ad_number_traits.h>
#include <deal.II/differentiation/ad/adolc_number_types.h>
#include <deal.II/differentiation/ad/adolc_product_types.h>
#include <deal.II/differentiation/ad/sacado_number_types.h>
#include <deal.II/differentiation/ad/sacado_product_types.h>
```
It also means that if we add support for other numbers then we only need to maintain a this single header (plus whatever lists are in the build system) in order to ensure the compatibility of these classes.
  
Do you agree with my decision to use the unifying header, or would we prefer to keep the individual headers in place?

Reply:
I am fine to keep a single one which has all AD types in.